src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,303 @@
+            int X2 = (short)_calibrationData.MC * (1 << 11) / (X1 + (short)_calibrationData.MD);
+            int B5 = (X1 + X2);
+
+            // Pressure Calculations
+            int B6 = B5 - 4000;
+            X1 = ((short)_calibrationData.B2 * (B6 * B6 / (1 << 12))) / (1 << 11);
+            X2 = (short)_calibrationData.AC2 * B6 / (1 << 11);
+            int X3 = X1 + X2;
+            int B3 = ((((short)_calibrationData.AC1 * 4 + X3) << (short)Sampling.Standard) + 2) / 4;
+            X1 = (short)_calibrationData.AC3 * B6 / (1 << 13);
+            X2 = ((short)_calibrationData.B1 * (B6 * B6 / (1 << 12))) / (1 << 16);
+            X3 = ((X1 + X2) + 2) / (1 << 2);
+            int B4 = _calibrationData.AC4 * (X3 + 32768) / (1 << 15);
+            int B7 = (UP - B3) * (50000 >> (short)Sampling.Standard);
+
+            int p = 0;
+            if((uint)B7 < 0x80000000)
+            {                
+                p = (int)(((long)B7 * 2) / B4);                
+            }
+            else
+            {               
+                p = (int)(((long)B7 / B4) * 2);             
+            }
+            
+            X1 = (p / (1 << 8)) * (p / (1 << 8));
+            X1 = (X1 * 3038) / (1 << 16);
+            X2 = (-7357 * p) / (1 << 16);
+            p = p + (X1 + X2 + 3791) / (1 << 4);
+
+            return p;
+        }
+
+        /// <summary>
+        ///  Calculates the altitude in meters from the specified sea-level pressure(in hPa).
+        /// </summary>
+        /// <param name="seaLevelPressure" > 
+        ///  Sea-level pressure in hPa
+        /// </param>
+        /// <returns>
+        ///  Height in meters from the sensor
+        /// </returns>
+        public double ReadAltitude(double sealevel_pa = 101325.0)
+        {
+            double pressure = (double)ReadPressure();
+            double altitude = 44330.0 * (1.0 - Math.Pow((pressure / sealevel_pa), (1.0 / 5.255)));
+
+
+            return altitude;
+        }
+
+        /// <summary>
+        ///  Calculates the pressure at sealevel when given a known altitude in meter
+        /// </summary>
+        /// <param name="altitude" > 
+        ///  altitude in meters
+        /// </param>
+        /// <returns>
+        ///  Pressure in Pascals
+        /// </returns>
+        public double ReadSeaLevelPressure(double altitude = 0.0)
+        {
+            double pressure = (double)ReadPressure();
+            double p0 = pressure / Math.Pow((1.0 - (altitude / 44333.0)), 5.255);
+
+            return p0;
+        }
+
+        /// <summary>
+        ///  Reads raw temperatue from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Raw temperature
+        /// </returns>
+        short ReadRawTemperature()
+        {
+            // Reads the raw (uncompensated) temperature from the sensor
+            _i2cDevice.Write(new[] { (byte)Register.CONTROL, (byte)Register.READTEMPCMD });            

```

Comment:
@joperezr, should we have Write API overloads on i2c that accept 1-4 bytes to avoid having the user create such arrays?

Reply:
I took these as example

https://github.com/dotnet/iot/blob/f45fda09308bc3d0c9d4c52fa1f02e1e8c31b71f/src/System.Device.Gpio/System/Device/I2c/Drivers/Windows10I2cDevice.Windows.cs#L77-L80

https://github.com/dotnet/iot/blob/d6808f0c2893ab2480636b5f25b2945c79fc6b3c/src/devices/Ds3231/Ds3231.cs#L106-L107

https://github.com/dotnet/iot/blob/789a72c11933d2e38cd87beb552b61dff249665c/src/devices/Ads1115/Ads1115.cs#L67-L68

https://github.com/dotnet/iot/blob/2055dc97a89b732f5f9d82dbcd9aa87734b4c8ec/src/devices/Bmp280/Bmp280.cs#L66

Reply:
I'd recommend using stackallock instead - I do not feel like we should have perf convenience APIs just yet

```csharp
Span<byte> foo = stackalloc byte[] { <your values> };
_i2cDevice.Write(foo);
```

You've happened to look at the bad examples :smile: